master/staging/README.md update

jiefeng-xu opened the pull request:
Update the staging repo list in README.md file to align with the directories under https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io. 

Leave the following repos as plain text as there are no external repos to link to yet.
`cli-runtime`
`kube-controller-manager`
`kube-proxy`
`kube-scheduler`
`kubelet`
`sample-cli-plugin`

Fixes #68235 

Signed-off-by: Stanton Xu <xjiefeng@gmail.com>

jiefeng-xu committed:
master/staging/README.md update

Update the staging repo list in README.md file to align with the 
directories under kubernetes/staging/src/k8s.io

Addressing issue https://github.com/kubernetes/kubernetes/issues/68235

Signed-off-by: Stanton Xu <xjiefeng@gmail.com>

stantonxu commented:
/sig architecture
/kind documentation

stantonxu commented:
/assign @dims

stantonxu commented:
/assign @neolit123

nikhita commented on staging/README.md:
```diff
@@ -6,15 +6,22 @@ top-level k8s.io repositories.
+- `k8s.io/kube-controller-manager`
+- `k8s.io/kube-proxy`
+- `k8s.io/kube-scheduler`
+- `k8s.io/kubelet`
```
@stantonxu Thanks for your first contribution to Kubernetes! :+1: 

@neolit123 @dims since we are already in code freeze, wdyt about waiting until the repos are created before adding them here? The request for creation already exist in https://github.com/kubernetes/org/issues/56 and https://github.com/kubernetes/org/issues/58.

nikhita commented:
Also, wdyt about adding a small note about the process for creating a new staging repo i.e. we include details about creating a request for creation of the org through sig-arch, then creating an issue on k/org and so on.

nikhita commented:
/release-note-none
/ok-to-test

dims commented:
+1 @nikhita ( adding a small note about the process for creating a new staging repo )

neolit123 commented:
ok @stantonxu,
as @nikhita and @dims have agreed, let's wait for the staging repositories to be created and we can then add the links in this PR.

hopefully this happens soon.

stantonxu commented:
Thank you @nikhita @neolit123 @dims for your comments. I agree with you, it makes sense to wait till the repos are created. I will update it with the links then.

stantonxu commented:
@spiffxp , welcome to comment, :-)

Please let me know when the repos are created, so that I will update this PR.

jiefeng-xu committed:
Add links to the newly created repos.
`cli-runtime`
`kube-controller-manager`
`kube-proxy`
`kube-scheduler`
`kubelet`
`sample-cli-plugin`

stantonxu commented:
@dims @spiffxp @nikhita @neolit123 I just noticed that the following external repos were created recently, so just pushed another commit to add the links to then. 

Now this PR is ready for review and merge. Thanks.

`cli-runtime`
`kube-controller-manager`
`kube-proxy`
`kube-scheduler`
`kubelet`
`sample-cli-plugin`

neolit123 commented:
thank you for the update and the patience @stantonxu 
i've verified all the links.

/lgtm

neolit123 commented:
/test pull-kubernetes-e2e-gce-device-plugin-gpu

stantonxu commented:
@neolit123 , I don't understand why the README update will cause the pull-kubernetes-e2e-gce-device-plugin-gpu test fail, which sounds like a function test. Is it a known issue?

neolit123 commented:
@stantonxu 
some k8s test can fail from time to time for no apparent reason - e.g. networking issues.
this is called a "flake".

stantonxu commented:
/test pull-kubernetes-e2e-kops-aws

neolit123 commented:
/test pull-kubernetes-e2e-kops-aws

nikhita commented:
@stantonxu thanks for working on this! :+1: 

This lgtm. We can add the repo creation guidelines as a follow-up.

>  Since I'm looking at staging repo creation right now I'll probably have something to say in a bit

@spiffxp any further comments?

spiffxp commented:
/milestone v1.12
/priority critical-urgent
SIG Testing chair, I'd like to make sure this readme lists the repos that are actually cut as part of v1.12

spiffxp commented:
/assign @smarterclayton @lavalamp 
for the /approve

spiffxp commented:
@nikhita This picked up all the staging repos I created, thanks for checking

My other thought is: this is error prone and busy work that could be automated away.  Another example of this pattern: https://github.com/kubernetes/k8s.io/blob/master/k8s.io/README.md is human-verified to be in sync with an nginx configmap.

lavalamp commented:
/approve

stantonxu commented:
/test pull-kubernetes-e2e-kops-aws

nikhita commented:
> My other thought is: this is error prone and busy work that could be automated away. Another example of this pattern: https://github.com/kubernetes/k8s.io/blob/master/k8s.io/README.md is human-verified to be in sync with an nginx configmap.

@spiffxp maybe this could be verified/synced up with the publishing bot config rules: https://github.com/kubernetes/publishing-bot/blob/master/configs/kubernetes-rules-configmap.yaml?

So the follow-up will need to include:

- repo creation methods
- a note about making sure that the list of repos are verified against the publishing bot


cc @sttts
